packages/manager-react-components/src/components/datagrid/datagrid.component.tsx

```diff
@@ -139,8 +174,64 @@ export const Datagrid = <T,>({
     }),
   });
 
+  const [columnsFilters, setColumnsFilters] = useState([]);
+  useEffect(() => {
+    const clmFilters = columns
+      .filter(
+        (item) =>
+          ('comparator' in item || 'type' in item) &&
+          'isFilterable' in item &&
+          item.isFilterable,
+      )
+      .map((column) => ({
+        id: column.id,
+        label: column.label,
+        ...(column?.type && { comparators: FilterCategories[column.type] }),
+        ...(column?.comparator && { comparators: column.comparator }),
+      }));
+    setColumnsFilters(clmFilters);
+  }, [columns]);
+
   return (
     <div>
+      {columnsFilters.length > 0 && (
+        <div className="flex flex-row-reverse py-[24px]">
+          <div id="datagrid-filter-popover-trigger">
+            <OdsButton
+              slot="datagrid-filter-popover-trigger"
+              size={ODS_BUTTON_SIZE.sm}
+              variant={ODS_BUTTON_VARIANT.ghost}
+              icon={ODS_ICON_NAME.filter}
+              label=""
+            />
```

Comment:
If we have icon with no label maybe add `aria-label` for visually impaired person ?